src/server/components/dashboard/controllers.ts

```diff
@@ -152,12 +157,66 @@ export async function listsController(req: Request, res: Response, next: NextFun
       isNewUser,
       lists,
       csrfToken: getCSRFToken(req),
+      dashboardBoxes: calculateDashboardBoxes(lists as ListsForDashboard[], !req.user?.isAdministrator),
     });
   } catch (error) {
     next(error);
   }
 }
 
+export function calculateSortOrder(
+  queryParamSortOrder: Prisma.ListsForDashboardOrderByWithRelationInput
+): Array<Record<string, string>> {
+  const defaultSortOrder = {
+    country: "asc",
+    type: "asc",
+  };
+
+  // TODO: Object.hasOwn is recommended but is not currently supported by tsc.
+  // eslint-disable-next-line no-prototype-builtins
+  if (!queryParamSortOrder.hasOwnProperty("admins")) {
+    return Object.entries(defaultSortOrder).map(convertEntryToObject);
+  }
+
+  const newSortOrder = {
+    ...queryParamSortOrder,
+    ...defaultSortOrder,
+  };
+
+  return Object.entries(newSortOrder).map(convertEntryToObject);
+}
+
+function convertEntryToObject([key, value]: [string, string]) {
+  return { [key]: value === "desc" ? "desc" : "asc" };
+}
+
+function calculateDashboardBoxes(lists: ListsForDashboard[], userIsAdmin: boolean) {
```

Comment:
I think you are passing in !req.user.isAdministrator. "not an admin". is that right?

Reply:
That's correct. It's more like `userRoleIsSuperAdmin` or something like that.

Reply:
Ah.. yeah.. getting confusing already. Let’s keep (as much as possible) administrator and user in the back end, and use ”admins” (to refer to regular users) on the frontend/templates only